ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -0,0 +1,287 @@
+import org.iplantc.de.client.models.ontologies.OntologyHierarchy;
+import org.iplantc.de.client.models.ontologies.OntologyMetadata;
+import org.iplantc.de.client.models.ontologies.OntologyVersionDetail;
+import org.iplantc.de.client.util.CommonModelUtils;
+import org.iplantc.de.commons.client.ErrorHandler;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+import org.iplantc.de.commons.client.info.SuccessAnnouncementConfig;
+
+import com.google.common.collect.Lists;
+import com.google.gwt.http.client.URL;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+import com.google.inject.Inject;
+
+import com.sencha.gxt.data.shared.ListStore;
+import com.sencha.gxt.data.shared.TreeStore;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author aramsey
+ */
+public class OntologiesPresenterImpl implements OntologiesView.Presenter,
+                                                ViewOntologyVersionEvent.ViewOntologyVersionEventHandler,
+                                                SelectOntologyVersionEvent.SelectOntologyVersionEventHandler,
+                                                SaveOntologyHierarchyEvent.SaveOntologyHierarchyEventHandler,
+                                                PublishOntologyClickEvent.PublishOntologyClickEventHandler,
+                                                HierarchySelectedEvent.HierarchySelectedEventHandler {
+
+    @Inject DEProperties properties;
+    @Inject IplantAnnouncer announcer;
+    private OntologiesView view;
+    private OntologyServiceFacade serviceFacade;
+    private final TreeStore<OntologyHierarchy> treeStore;
+    private OntologiesView.OntologiesViewAppearance appearance;
+    private AdminCategoriesView.Presenter categoriesPresenter;
+    private AdminAppsGridView.Presenter gridPresenter;
+    private OntologyAutoBeanFactory beanFactory;
+    ListStore<App> listStore;
+    private String UNCLASSIFIED_LABEL = "Unclassified";
+    private String UNCLASSIFIED_IRI_APPEND = "_unclassified";
+
+    @Inject
+    public OntologiesPresenterImpl(OntologyServiceFacade serviceFacade,
+                                   final TreeStore<OntologyHierarchy> treeStore,
+                                   OntologyAutoBeanFactory beanFactory,
+                                   OntologiesViewFactory factory,
+                                   OntologiesView.OntologiesViewAppearance appearance,
+                                   AdminCategoriesView.Presenter categoriesPresenter,
+                                   AdminAppsGridView.Presenter gridPresenter) {
+        this.serviceFacade = serviceFacade;
+        this.beanFactory = beanFactory;
+        this.treeStore = treeStore;
+        this.appearance = appearance;
+
+        this.categoriesPresenter = categoriesPresenter;
+        this.gridPresenter = gridPresenter;
+        listStore = gridPresenter.getView().getGrid().getStore();
```

Comment:
We should try avoid accessing view elements in presenter. Instead call a method  on view that does the operation. so instead of listStore.clear() you will call view.clear() which in turn clears the store.